code/controllers/subsystem/puddle.dm

```diff
@@ -0,0 +1,17 @@
+SUBSYSTEM_DEF(puddle)
+	name = "Puddle"
+	wait = 5
+	runlevels = RUNLEVEL_GAME
+	var/list/puddlelist = list()
+
+/datum/controller/subsystem/puddle/fire(resumed)
+	set waitfor = FALSE
+	var/list/puddles = list()
+	for(var/obj/puddle/puddle in puddlelist)
+		puddles |= puddle
```

Comment:
Why?

Reply:
Because it's modifying the list as it iterates, didn't want it to cause problems